prisma/schema.prisma

```diff
@@ -76,28 +76,54 @@ model Org {
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
   users     Member[]
+  roles     Role[]
 }
 
-enum AccessEnum {
-  READ
-  WRITE
-  ADMIN
+enum MEMBERHIP_STATUS {
+  ACCEPTED
+  PENDING
+  DECLINED
 }
 
 model Member {
-  id        String   @id @default(cuid())
-  title     String?
-  access    AccessEnum @default(READ)
-  active    Boolean  @default(true)
+  id           String           @id @default(cuid())
+  title        String?
+  active       Boolean          @default(true)
+  status       MEMBERHIP_STATUS @default(PENDING)
+  // When the user joins, we will clear out the name and email and set the user.
+  invitedName  String?
+  invitedEmail String?
+
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
-  
-  userId    String?
-  user      User?    @relation(fields: [userId], references: [id])
 
-  orgId     String?
-  org       Org?     @relation(fields: [orgId], references: [id])
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id])
+
+  orgId String?
+  org   Org?    @relation(fields: [orgId], references: [id])
+
+  roleId Int
+  role   Role @relation(fields: [roleId], references: [id])
+
+  @@unique([userId, orgId, invitedEmail])
+  @@index([orgId])
+  @@index([roleId])
+}
+
+model Role {
```

Comment:
added model Role where we can custom roles and default ones.

whenever a org is created we can create some default roles , which can be used by  https://casl.js.org/v6/en/ to manage permissions

```
db.role.createMany({
  data: [
    {
      name: "admin",
      orgId: "sample-org",
      permissions: [{ action: "manage", subject: "all" }],
    },
    {
      name: "member",
      orgId: "sample-org",
      permissions: [
        { action: "read", subject: "Article" },
        { action: "edit", subject: "Article" },
      ],
    },
  ],
})
```

Reply:
Thanks for opening this PR @G3root CASL is amazing, i love it and we will def use it or something similar when there is a need. For right now though, it will be an overkill. Thats why we want to go with something simple for now. Just three access levels (`read`, `write` and `admin` or `owner`. We will introduce roles table with proper RBAC if/when customer ask or we realize that we need for sure.